source/extensions/common/wasm/exports.cc

```diff
@@ -498,6 +499,48 @@ Word get_buffer_status(void* raw_context, Word type, Word length_ptr, Word flags
   return wasmResultToWord(WasmResult::Ok);
 }
 
+Word set_buffer(void* raw_context, Word type, Word start, Word length, Word data_ptr,
+                Word data_size, Word flags) {
+  // TODO(PiotrSikora): use flags.
+  UNREFERENCED_PARAMETER(flags);
+  if (type.u64_ > static_cast<uint64_t>(BufferType::MAX)) {
+    return wasmResultToWord(WasmResult::BadArgument);
+  }
+  auto context = WASM_CONTEXT(raw_context);
+  auto buffer = context->getBuffer(static_cast<BufferType>(type.u64_));
+  if (!buffer) {
+    return wasmResultToWord(WasmResult::NotFound);
+  }
+  auto data = context->wasmVm()->getMemory(data_ptr.u64_, data_size.u64_);
+  if (!data) {
+    return wasmResultToWord(WasmResult::InvalidMemoryAccess);
+  }
+  WasmResult result = WasmResult::InternalFailure;
+  if (context->setBuffer(static_cast<BufferType>(type.u64_),
```

Comment:
This should just be a thin wrapper to get the string_view(s) and do other format conversions then call a function on Context to do the real work.